sklearn/model_selection/tests/test_successive_halving.py

```diff
@@ -52,6 +52,71 @@ def get_params(self, deep=False):
         return params
 
 
+class SometimesFailClassifier(DummyClassifier):
+    def __init__(
+        self,
+        strategy="stratified",
+        random_state=None,
+        constant=None,
+        n_estimators=10,
+        fail_fit=False,
+        fail_predict=False,
+        a=0,
+    ):
+        self.fail_fit = fail_fit
+        self.fail_predict = fail_predict
+        self.n_estimators = n_estimators
+        self.a = a
+
+        super().__init__(
+            strategy=strategy, random_state=random_state, constant=constant
+        )
+
+    def fit(self, X, y):
+        if self.fail_fit:
+            raise Exception("fitting failed")
+        return super().fit(X, y)
+
+    def predict(self, X):
+        if self.fail_predict:
+            raise Exception("predict failed")
+        return super().predict(X)
+
+
+@pytest.mark.parametrize("Est", (HalvingGridSearchCV, HalvingRandomSearchCV))
```

Comment:
There are warnings that are expected because `SometimesFailClassifier` is expected to fail:

```suggestion
@pytest.mark.filterwarnings("ignore::sklearn.exceptions.FitFailedWarning")
@pytest.mark.filterwarnings("ignore::UserWarning")
@pytest.mark.parametrize("Est", (HalvingGridSearchCV, HalvingRandomSearchCV))
```

Reply:
How did you see these warnings? When I run `pytest -svk nan_handling sklearn/model_selection/tests/test_successive_halving.py` it just shows the succesfull tests and no warnings. I thought `-v` would get me "all the output" to see stuff like this :-/

Reply:
You can turn all warnings into errors:

```bash
pytest sklearn/model_selection/tests/test_successive_halving.py::test_nan_handling -Werror
```

We can clarify which warnings we are ignoring like so:

```python
@pytest.mark.filterwarnings("ignore::sklearn.exceptions.FitFailedWarning")
@pytest.mark.filterwarnings("ignore:Scoring failed:UserWarning")
@pytest.mark.filterwarnings("ignore:One or more of the:UserWarning")
```

Reply:
Thanks! And applied